src/Functions/DivisionUtils.h

```diff
@@ -178,4 +178,21 @@ struct ModuloLegacyImpl : ModuloImpl<A, B>
     using ResultType = typename NumberTraits::ResultOfModuloLegacy<A, B>::Type;
 };
 
+template <typename A, typename B>
+struct PositiveModuloImpl : ModuloImpl<A, B>
+{
+    using OriginResultType = typename ModuloImpl<A, B>::ResultType;
+    using ResultType = typename NumberTraits::ResultOfPositiveModulo<A, B>::Type;
+
+    template <typename Result = ResultType>
+    static inline Result apply(A a, B b)
+    {
+        auto res = ModuloImpl<A, B>::template apply<OriginResultType>(a, b);
+        if constexpr (is_signed_v<A>)
+            if (res < 0)
```

Comment:
I guess this function should be slower than `modulo()` because of this `if` (it should prevent vectorisation). let's add a perf test comparing them and if it is the case, let's add a note in docs that `modulo` should be used when possible.

Reply:
Good advice, I will test it.